edx_event_bus_kafka/internal/consumer.py

```diff
@@ -179,18 +192,32 @@ def reset_offsets(consumer, partitions):
             self.consumer.subscribe([full_topic], on_assign=reset_offsets)
             logger.info(f"Running consumer for {run_context!r}")
 
+            # How many errors have we seen in a row? If this climbs too high, exit with error.
+            # Any error counts, here—whether due to a polling failure or a message processing
+            # failure. But only a successfully processed message clears the counter. Just
+            # being able to talk to the broker and get a message (or a normal poll timeout) is
+            # not sufficient to show that progress can be made.
+            consecutive_errors = 0
+
             while True:
                 # Allow unit tests to break out of loop
                 if self._shut_down_loop:
                     break
 
+                # Detect probably-broken consumer and exit with error.
```

Comment:
Nit: My brain would put this at the end of the loop, but I guess yours wouldn't. :)

Reply:
I remember considering whether I wanted it at the beginning or the end, but I don't remember why I chose this. 🤔 I suspect it's because of the general principle that there can be multiple ways to exit an iteration (`continue` or falling through), but only one way to enter one. There _isn't_ a `continue` statement here, so it could go at the end as well. The other likely reason is that I wanted it near the `_shut_down_loop`, thematically. :-) But neither is a particularly overwhelming reason.